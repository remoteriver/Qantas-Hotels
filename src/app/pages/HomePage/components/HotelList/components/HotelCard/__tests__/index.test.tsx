import { render } from '@testing-library/react';
import { createRenderer } from 'react-test-renderer/shallow';
import {
  CancellationType,
  Currency,
  ImageType,
  PromotionType,
  RatingType,
} from 'types/Hotel';
import { HotelCard } from '..';

const renderer = createRenderer();

describe('<HotelCard />', () => {
  const mockHotel = {
    id: 'cxd650nuyo',
    property: {
      propertyId: 'P107801',
      title: 'Courtyard by Marriott Sydney-North Ryde',
      address: ['7-11 Talavera Rd', 'North Ryde'],
      previewImage: {
        url: 'https://unsplash.it/145/125/?random',
        caption: 'Image of Courtyard by Marriott Sydney-North Ryde',
        imageType: ImageType.PRIMARY,
      },
      rating: {
        ratingValue: 4.5,
        ratingType: RatingType.SELF,
      },
    },
    offer: {
      promotion: {
        title: 'Exclusive Deal',
        type: PromotionType.MEMBER,
      },
      name: 'Deluxe Balcony Room',
      displayPrice: {
        amount: 329.0,
        currency: Currency.AUD,
      },
      savings: {
        amount: 30.0,
        currency: Currency.AUD,
      },
      cancellationOption: {
        cancellationType: CancellationType.NOT_REFUNDABLE,
      },
    },
  };

  const mockFn = jest.fn();

  beforeAll(() => {});

  it('should render and match the snapshot', () => {
    //act
    renderer.render(
      <HotelCard hotel={mockHotel} removeHandler={mockFn}></HotelCard>,
    );
    const renderedOutput = renderer.getRenderOutput();

    //assert
    expect(renderedOutput).toMatchSnapshot();
  });
});
