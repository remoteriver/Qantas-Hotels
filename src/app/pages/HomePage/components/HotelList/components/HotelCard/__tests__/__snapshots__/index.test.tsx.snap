// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HotelCard /> should render and match the snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  className="makeStyles-root-1"
  container={true}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <div
      className="makeStyles-imgContainer-2"
    >
      <div
        className="makeStyles-banner-3"
      >
        Exclusive Deal
      </div>
      <img
        alt="hotel preview logo"
        src="https://unsplash.it/145/125/?random"
      />
    </div>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-content-4"
    container={true}
    item={true}
    xs={true}
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      direction="column"
      item={true}
      xs={10}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        item={true}
      >
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-propertyTitle-5"
          variant="h5"
        >
          Courtyard by Marriott Sydney-North Ryde
        </WithStyles(ForwardRef(Typography))>
        <RatingBar
          rating={
            Object {
              "ratingType": "self",
              "ratingValue": 4.5,
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Typography))
          variant="caption"
        >
          7-11 Talavera Rd
          North Ryde
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <Styled(MuiBox)
          marginY={1}
        >
          <WithStyles(ForwardRef(Link))>
            <WithStyles(ForwardRef(Typography))
              variant="caption"
            >
              Deluxe Balcony Room
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(Link))>
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      direction="column"
      item={true}
      xs={2}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      />
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <Styled(MuiBox)
          textAlign="right"
        >
          <WithStyles(ForwardRef(Typography))
            component="span"
            variant="body2"
          >
            1
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            component="span"
            variant="caption"
          >
             night total (AUD)
          </WithStyles(ForwardRef(Typography))>
        </Styled(MuiBox)>
        <Styled(MuiBox)
          textAlign="right"
        >
          <WithStyles(ForwardRef(Grid))
            alignItems="flex-start"
            container={true}
            justifyContent="flex-end"
            spacing={1}
          >
            <WithStyles(ForwardRef(Grid))
              item={true}
            >
              <WithStyles(ForwardRef(Typography))
                component="span"
                variant="body2"
              >
                $
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(Grid))>
            <WithStyles(ForwardRef(Grid))
              item={true}
            >
              <WithStyles(ForwardRef(Typography))
                component="span"
                variant="h5"
              >
                329
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(Grid))>
          </WithStyles(ForwardRef(Grid))>
        </Styled(MuiBox)>
        <Styled(MuiBox)
          textAlign="right"
        >
          <WithStyles(ForwardRef(Typography))
            component="span"
            variant="body1"
          >
            Save $30~
          </WithStyles(ForwardRef(Typography))>
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
